### ISSUE TITLE
[Issue]: 5.1 surround sound does not work

### ISSUE THREAD

**Issue by smirnovv2007** [ClassicOldSong/moonlight-android#22]
### Describe the bug

I tried to run Artemis on the Mi TV Box S 2nd gen connected to Samsung home theater with 5.1 sound, but it sounds only stereo. In Artemis settings, 5.1 sound is set, in Windows too.

### Steps to reproduce

-

### Other Moonlight clients

None

### Moonlight adjusted settings

No

### Moonlight default settings

Yes

### Gamepad-related connection issue

No

### Gamepad-related input issue

Yes

### Gamepad-related streaming issue

Yes

### Android version

Android 11 TV

### Device model

mi tv box s 2nd gen

### Server PC OS version

Windows 11 23H2

### Server PC GeForce Experience version

Latest [Apollo](https://github.com/ClassicOldSong/Apollo)

**Comment by ClassicOldSong** [comment]
Does original Moonlight and Sunshine have the same issue? I haven't touched the code about surround sound so this might be a problem from upstream.

**Comment by smirnovv2007** [comment]
> Does original Moonlight and Sunshine have the same issue?

Yes, it does(

**Comment by christopher-s** [comment]
Of note, I am not encountering this issue with my setup. Streaming from PC to Nvidia Shield - I have 5.1 surround sound in everything I've tested.

**Comment by kroesufos** [comment]
> Of note, I am not encountering this issue with my setup. Streaming from PC to Nvidia Shield - I have 5.1 surround sound in everything I've tested.

Same, just more audio delay

**Comment by JohnnyPh** [comment]
On my new Onn 4K Pro, I can get 5.1 audio working, but the channels keep swapping at random—sometimes the subwoofer shifts to the rear-right speaker, the front channel jumps to the rear-left, and they just keep flipping back and forth.

**Comment by alxtraxxx** [comment]
I can confirm OG moonlight does not play surround with more than Left + Right channel on various devices.
Funny thing is on legacy HDMI hardware it works correctly (HDMI <2.1 with lower HDCP, not sure if HDCP is a part of the problem).
I have an old FHD AVR and if I connect my FireTV stick to it I get correct surround, if I connect my Stick to my 4k TV or 8k AVR I only get stereo again. Perhaps there is something wrong with the HDMI handshake.
I am using Steam Streaming Speakers and they work fine using Steam Link on all my android devices

**Comment by JohnnyPh** [comment]
> I can confirm OG moonlight does not play surround with more than Left + Right channel on various devices. Funny thing is on legacy HDMI hardware it works correctly (HDMI <2.1 with lower HDCP, not sure if HDCP is a part of the problem). I have an old FHD AVR and if I connect my FireTV stick to it I get correct surround, if I connect my Stick to my 4k TV or 8k AVR I only get stereo again. Perhaps there is something wrong with the HDMI handshake. I am using Steam Streaming Speakers and they work fine using Steam Link on all my android devices

I found out the Apollo pork work great with 5.1 channels. I don’t give a link here but you know what I mean.

**Comment by alxtraxxx** [comment]
Just checked artemis + apollo and it receives only front left + front right channel like Sunshine/OG moonlight. I am not talking about already DTS encoded videos, I am talking about real time 7.1 in games.My receiver only gets PCM Front Left + Right, if you dont have a AVR that shows you input signal params use windows speaker configuration, set it to 5.1 and run test, there will be no sound from the other speakers if you havent upmixing etc
 enabled.
Edit: As I said and others confirmed already it seems to be dependend on HDMI handshake on init as older HDMI devices work on client side

**Comment by JohnnyPh** [comment]
I use Artemis and Apollo combo and it is working correctly in my game. Did you turn on the 5.1 or 7.1 setting in the android app client? Because it is defaut to stereo, and you need to also change your sound setting on windows host to 5.1 or 7.1 (it is the stream audio or something like that).

**Comment by alxtraxxx** [comment]
Yes and it does not work. Windows is configured correctly, sunshine/artemis/apollo/moonlight too but client only sends FR+FL. If I move same FireTV stick into my old FHD AVR which is <HDMI 2.1 and HDCP 1.3 only handshake screen while connecting to host looks different and surround works. And it is not a problem of my 8k AVR, on any of my Google/Android TVs its only stereo. If I use windows client of moonlight it works too. So the issue is definately part of the audio code used in moonlight and artemis. My TV and my AVR support ANY audio format currently available and even all video formats so thats not the problem.

**Comment by alxtraxxx** [comment]
Update: For Sunshine + Moonlight I found a way to make it work (currenly using Sunshine Version 2025.822.34814),
I forced sunshine to use 7.1 my Audio Device as Audio sink, Steam Streaming Speakers as Virtual Sink, forced Video to my
virtual display and set HDR workaround to 500ms. After doing so, connetion screen\messages in moonlight take longer instead of
almost immidiately connecting and I get full 7.1 surround again. So maybe the same bug when host is capturing from wrong device even if windows and logs say it would use the correct audio device. Hope this will help people with the same problems, not sure if its always the same issues

**Comment by alxtraxxx** [comment]
Update again: Found a 100% fix for me on Fire TV Stick 4k max: Force resolution to 4k instead auto, switch HDR to anything and then 12 bit again (max for my setup) and it will connect with surround. My guess is when the android device is set to auto resolution etc. it tries to switch resolution and by doing this it kills of the correct hdmi parameters of the handshake so it runs only stereo on receiving device. If I force the stick to keep res I can read all connection messages of the client (instead of getting a short blank screen) and get surround. As not all devices (eg Android TVs) can be forced to keep a resolution/dont try to reset I am not sure how to fix that on those devices, not sure why res change is messing hdmi infos about sound channels.

**Comment by JohnnyPh** [comment]
> Update again: Found a 100% fix for me on Fire TV Stick 4k max: Force resolution to 4k instead auto, switch HDR to anything and then 12 bit again (max for my setup) and it will connect with surround. My guess is when the android device is set to auto resolution etc. it tries to switch resolution and by doing this it kills of the correct hdmi parameters of the handshake so it runs only stereo on receiving device. If I force the stick to keep res I can read all connection messages of the client (instead of getting a short blank screen) and get surround. As not all devices (eg Android TVs) can be forced to keep a resolution/dont try to reset I am not sure how to fix that on those devices, not sure why res change is messing hdmi infos about sound channels.

Not true, my Firestick 4k Max 2nd Gen setting are below and works fine.
Video resolution: Auto
Matching fram rate: Off
Color Depth: Up to 12 bits
Color Format: RGB
Dynamic Range Setting: Always HDR

**Comment by alxtraxxx** [comment]
If your FireTV is connected to old HDMI/HDCP perhaps, on my old FHD AVR it worked too but not on my 4k HDR 144 Hz ALLM TV, has the same problems on my Android TV itself, TV stick is connected to my 8k AVR behaves the same, while it worked on my old AVR. There is definately a difference when connecting to stream with other settings. But why are you forcing HDR on SDR content too? HDR does look worse on my streaming client with Mini LED TV than on my host with IPS so I keep it now at SDR for moonlight and else Auto depending if source is HDR or not. Color format RGB does not work for me as I have a 7900XTX which can not capture 4k HDR with RGB, only the newer ones can

**Comment by JohnnyPh** [comment]
I dont know how do you defind “old” but my tv is Samsung Q60D 2025 (entry level of samsung brand), and pair with q600c 5.1 setup.

I forcing HDR because I don’t like a short blackscreen when tv switching between modes. And my tv doesn’t behave anything different when I force it.

**Comment by alxtraxxx** [comment]
Your TV has only HDMI 2.0, mine 2.1, ALLM and VRR, might be the difference, I have no idea. I can only say the workaround works for me on different devices. Artemis behaves the same. I am just happy I found a workaround now that works everytime for me. My FireTV stick is connected to Denon AVR-S770H which is connected to my 144Hz eArc VRR ALLM HDMI 2.1 port, AVR is set to HDMI enhanced 8k, but makes no difference at 4k as the FireTV cant do more than 60Hz. If FireTV is connected to TV directly it still needs the workaround.
